library/src/main/java/com/chuckerteam/chucker/internal/ui/transaction/TransactionPayloadFragment.kt

```diff
@@ -71,21 +71,50 @@ internal class TransactionPayloadFragment :
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+
+        val result = mutableListOf<TransactionPayloadItem>()
+        val adapter = TransactionBodyAdapter(result)
+        payloadBinding.responseRecyclerView.adapter = adapter
+        payloadBinding.responseRecyclerView.setHasFixedSize(true)
+
+        viewModel.encodeUrl.observe(
+            viewLifecycleOwner,
+            Observer { encode ->
+                if (adapter.itemCount == 0) {
+                    return@Observer
+                }
+                val transaction = viewModel.transaction.value ?: return@Observer
+                uiScope.launch {
+                    updateUpdater(result, adapter, transaction, !encode)
+                }
+            }
+        )
+
         viewModel.transaction.observe(
             viewLifecycleOwner,
             Observer { transaction ->
                 if (transaction == null) return@Observer
                 uiScope.launch {
                     showProgress()
-                    val result = processPayload(type, transaction)
-                    payloadBinding.responseRecyclerView.adapter = TransactionBodyAdapter(result)
-                    payloadBinding.responseRecyclerView.setHasFixedSize(true)
+                    val encode = viewModel.encodeUrl.value ?: true
```

Comment:
Why don't you call this variable the same way as you named parameter `updateAdapter` fun?

Reply:
As I mentioned, I need to rename some stuff. I started with just the formatting of url encoded forms and ended up with more. I was wondering if I should stick with just the formatting of url encoded forms or continue with this more generic solution.